semgrep-core/src/analyzing/AST_to_IL.ml

```diff
@@ -164,23 +164,23 @@ let add_instr env instr = Common.push (mk_s (Instr instr)) env.stmts
 
 (* Create an auxiliary variable for an expression---unless the expression
  * itself is already a variable! *)
-let mk_aux_var ?var env tok exp =
+let mk_aux_var ?str env tok exp =
   match exp.e with
   | Fetch ({ base = Var var; offset = NoOffset; _ } as lval) -> (var, lval)
   | _ ->
-      let var = fresh_var ?var env tok in
+      let var = fresh_var ?str env tok in
       let lval = lval_of_base (Var var) in
-      add_instr env (mk_i (Assign (lval, exp)) exp.eorig);
+      add_instr env (mk_i (Assign (lval, exp)) NoOrig);
```

Comment:
I will add a comment to IL.ml on how to use `Related` and `NoOrig`, although I don't know what are the right guidelines yet. In this particular case I put a `NoOrig` because I know that the RHS of the assignment already captures the correct `eorig`. If there was no subexpression carrying the origin then I would have added a `related_tok`. I try to avoid too many `related_exp` because it makes `-dump_il` harder to read. Although `related_tok` is lighter in that sense. In this case adding a `related_tok` would be harmless but in general those `tok` will be used by taint-tracking to decide whether something is a source/sanitized/sink.